Sources/DrawingTools/NativeImage Extensions.swift

```diff
@@ -0,0 +1,379 @@
+    ///                Defaults to `true`.
+    ///   - operation: The contextualized operation to perform while this image has focus lock
+    ///
+    /// - Returns: Anything the given function throws
+    ///
+    /// - Throws: Anything the given function throws
+    func inGraphicsContext<Return>(
+        _ context: GraphicsContext,
+        flipped: Bool = defaultFlipped,
+        withFocus: Bool = true,
+        do operation: OperationInGraphicsContextWithImage<Return>)
+        rethrows -> Return
+    {
+        func handleContextSwitch(_ image: NativeImage) throws -> Return {
+            try context.inCgContext { context in
+                try operation(image, context)
+            }
+        }
+        
+        if withFocus {
+            return try self.withFocus(flipped: flipped, do: handleContextSwitch)
+        }
+        else {
+            return try handleContextSwitch(self)
+        }
+    }
+    
+    
+    /// Allows you to perform a contextualized draw operation on this image with a the given context.
+    ///
+    /// - Parameters:
+    ///   - context:   The graphics context in which to run the given function
+    ///   - flipped:   _optional_ - Whether to flip the Y axis of the graphics context. Defaults to `defaultFlipped`.
+    ///   - withFocus: _optional_ - Whether to lock focus on the current image before entering the given function.
+    ///                Focus is guaranteed to be unlocked after `operation` exits.
+    ///                Defaults to `true`.
+    ///   - operation: The contextualized operation to perform while this image has focus lock
+    ///
+    /// - Returns: Anything the given function throws
+    ///
+    /// - Throws: Anything the given function throws
+    func inGraphicsContext<Return>(
+        _ context: GraphicsContext,
+        flipped: Bool = defaultFlipped,
+        withFocus: Bool = true,
+        do operation: OperationInGraphicsContext<Return>)
+        rethrows -> Return
+    {
+        try inGraphicsContext(
+            context,
+            flipped: flipped,
+            withFocus: withFocus) { _, context in
+            try operation(context)
+        }
+    }
+}
+
+
+
+public extension NativeImage {
+    
+    /// Creates a new image and immediately starts drawing on it
+    ///
+    /// - Note: There are two forms of this function: one that passes the `artist` an image and a context, and one that
+    ///         just passes it a context. This is the one which passes both.
+    ///
+    /// - Parameters:
+    ///   - size:    The size of the new image
+    ///   - context: _optional_ - The context in which to draw the new image
+    ///   - artist:  The function which will draw on the new image
+    ///
+    /// - Throws: Any error that `artist` throws
+    static func drawNew(
+        size: CGSize,
+        context: GraphicsContext = .current,
+        artist: OperationInGraphicsContextWithImage<Void>)
```

Comment:
Maybe make this a custom, specific typealias. Otherwise it's ambiguous what `Void` is for.